run_electrum

```diff
@@ -152,17 +152,13 @@ def init_cmdline(config_options, wallet_path, server, *, config: 'SimpleConfig')
         print_stderr("Exposing a single private key can compromise your entire wallet!")
         print_stderr("In particular, DO NOT use 'redeem private key' services proposed by third parties.")
 
-    # will we need a password
-    if not storage.is_encrypted():
-        db = WalletDB(storage.read(), manual_upgrades=False)
```

Comment:
This is not the only place where we load the WalletDB. There is similar logic in the qml/kivy GUIs, and notably in `run_offline_command`. Running `run_offline_command` while another daemon has the wallet file open can result in corruption. Some more generic protection against that would be desirable.

Reply:
The storage object is now passed to the `WalletDB` constructor:
```python 
db = WalletDB(storage.read(), storage=storage, upgrade=upgrade)
```

If we need to load a `WalletDB` for the sole purpose of checking whether it has a password, or if it requires upgrades, I think we should open it in a storage-less mode, which is possible with:
```python 
db = WalletDB(storage.read(), storage=None, upgrade=upgrade)
```